<h1 style="background-color:#3b5998;font-family:newtimeroman;font-size:300%;text-align:center;color:#FFFFFF">REGRESYON</h1>


<h1 style="background-color:#3b5998;font-family:newtimeroman;font-size:300%;text-align:left;color:#FFFFFF">REGRESYON NEDİR ? </h1>

Regresyon, istatistiksel bir analiz yöntemidir ve bir değişkenin diğer değişkenlerle olan ilişkisini anlamak, bu ilişkiyi modellemek ve gelecekteki değerleri tahmin etmek için kullanılır. Temel olarak, regresyon, bağımlı bir değişkenin bağımsız değişkenlerle nasıl ilişkilendiğini belirlemek için kullanılır. Bu bağlamda, bağımlı değişken genellikle tahmin yapmak istediğiniz şeyi temsil eder, bağımsız değişkenler ise bu tahmin üzerinde etkisi olduğunu düşündüğünüz faktörleri temsil eder.

Regresyon analizi, birçok farklı disiplinde kullanılır, örneğin:

Ekonomi ve Finans: Ekonomik değişkenler arasındaki ilişkileri anlamak ve gelecekteki ekonomik olayları tahmin etmek için kullanılır.

Pazarlama: Reklam harcamaları, fiyatlandırma ve diğer pazarlama stratejileri ile satışlar arasındaki ilişkileri anlamak ve optimize etmek için kullanılır.

Tıp: Hastalıkların risk faktörlerini belirlemek, bir tedavi yönteminin etkisini değerlendirmek veya hastalık prognozları yapmak için kullanılır.

Mühendislik: Bir değişkenin diğer değişkenlere olan etkisini anlamak ve bir sistemin performansını tahmin etmek için kullanılır.

Sosyal Bilimler: Toplumsal faktörlerin bireylerin davranışları üzerindeki etkilerini incelemek için kullanılır.




##### Örneğin;

Oda sayısı ve metrekare değişkenleri olan bir veri setini regresyon kullanarak modellenmesi.


y = $\beta_0 + \beta_1X + \beta_2X  + \varepsilon$

y = Ev Fiyatı

$\beta_0$ = Bağımsız değişkenlerimiz 0 olduğu zamanki değerimiz 50.000 olsun.

$\beta_1$ = Oda Sayısı 

$\beta_2$ = Metrekare 

$\varepsilon$ = Gerçek değerler ile tahmin edilen değerler arasındaki fark


Ev Fiyatı = 50.000 + 30 x Oda Sayısı + 20 x Metrekare + $\varepsilon$
 
 
($\beta_1$) Oda sayısı değişkeninin değeri 4 , ($\beta_2$) metrekare değişkeninin değeri 100 olsun .

Ev Fiyatı = 50.000 + 30x4 + 20x100 = 52.120 birim olarak tahmin edilir.

<h1 style="background-color:#FFA500;font-family:newtimeroman;font-size:200%;text-align:left;color:#FFFFFF"> Bağımlı Değişken (Y):
</h1>


Bu, tahmin yapmaya çalıştığınız veya açıklamak istediğiniz değişkendir. Bu değişken, diğer bağımsız değişkenler tarafından etkilenebilen ve model tarafından tahmin edilen değişkendir. Örneğin, bir satış tahmin modelinde satış miktarı bağımlı bir değişkendir.

<h1 style="background-color:#FFA500;font-family:newtimeroman;font-size:200%;text-align:left;color:#FFFFFF"> Bağımsız Değişken (X): 
</h1>


Bu, bağımlı değişkeni açıklamak veya tahmin etmek için kullanılan değişkenlerdir. Bağımsız değişkenlerin bağımlı değişken üzerinde bir etkisi olduğu düşünülmektedir. Örneğin, bir satış tahmin modelinde reklam harcamaları, fiyat ve promosyonlar bağımsız değişkenler olabilir.

<h1 style="background-color:#FFA500;font-family:newtimeroman;font-size:200%;text-align:left;color:#FFFFFF"> Kesim noktası  𝛽0
  Nedir ?
</h1>


Bağımsız değişkenlerimiz yani X'lerimiz 0 olduğunda bağımlı değişkenimizin beklenen değeridir.

Diyelim ki regresyon modeliniz ev fiyatlarını tahmin ediyorsa ve kesim noktası 50,000 TL ise bu durumda, hiçbir ek özellik veya faktör olmadan bir evin başlangıç fiyatının 50,000 TL olduğunu söyleyebiliriz. Yani, diğer faktörlerin etkisi olmasa bile, ev fiyatları bu değerle başlar.

<h1 style="background-color:#FFA500;font-family:newtimeroman;font-size:200%;text-align:left;color:#FFFFFF">Regresyon Katsayıları $\beta_1,\beta_2,\beta_3\,\beta_n$ Nedir ?
</h1>


Regresyon katsayıları bağımsız değikeninin bağımlı değişken üzerindeki etkilerini gösterir . 

Örneğin bir regresyon modeliniz var ve bu model öğrencinin sınav notunu a sayısındaki çalışma saatiyle tahmin ediyorsa bu durumda $\beta_1$ çalışma saatinin sınav notuna olan etkisini gösterir.Eğer $\beta_1$ = 2 ise her ekstra çalışma saatinin sınav notunda 2 birimlik bir artış olduğunu gösterir.

<h1 style="background-color:#FFA500;font-family:newtimeroman;font-size:200%;text-align:left;color:#FFFFFF"> Hata Terimi ( $\varepsilon$)
</h1>


#### $\varepsilon_i = Y_i - \hat{Y}_i$

$Y_i$ = Gerçek değerler

$\hat{Y}_i$ = Tahmin edilen değerleri temsil eder.


<h1 style="background-color:#3b5998;font-family:newtimeroman;font-size:300%;text-align:left;color:#FFFFFF">REGRESYON MODELLERİ
</h1>


<h1 style="background-color:#3b5998;font-family:newtimeroman;font-size:250%;text-align:center;color:#FFFFFF">Basit Lineer Regresyon 
</h1>

Basit lineer regresyon, yalnızca bir bağımsız değişkenin kullanıldığı bir regresyon türüdür. Bu model, bir bağımlı değişkenin, tek bir bağımsız değişkenle doğrusal bir ilişkisini ifade eder. Aşağıda temel matematiksel formül ve bu formüldeki terimlerin açıklamaları bulunmaktadır.

Y = $\beta_0 + \beta_1X + \varepsilon$

Y = Bağımlı değişken (Tahmin edilmeye çalışılan değişken)

$\beta_0$ = Kesim Noktası(Intercept)

$\beta_1$ = Eğim (Regresyon Katsayısı)

$\varepsilon$ = Hata Terimi

Her bir X değişkeni, bağımlı değişken Y üzerinde belirli bir katsayı ile çarpılarak toplanır. Bu, çoklu değişkenlerin etkileşimini modelleyerek daha karmaşık ilişkileri incelememize olanak tanır.

<h1 style="background-color:#3b5998;font-family:newtimeroman;font-size:250%;text-align:center;color:#FFFFFF">Çoklu Lineer Regresyon
</h1>

Y = $\beta_0 + \beta_1X + \beta_2X + ..\beta_nX + \varepsilon$

Y = Bağımlı değişken (Tahmin edilmeye çalışılan değişken)

$\beta_0$ = Kesim Noktası(Intercept)

$\beta_1,\beta_2,\beta_3\,\beta_n$ = Eğim (Regresyon Katsayısı)

$\varepsilon$ = Hata Terimi

 Çoklu lineer regresyon, birden fazla bağımsız değişkenin kullanıldığı bir regresyon türüdür. Bu model, bir bağımlı değişkenin birden fazla bağımsız değişkenle doğrusal bir ilişkisini ifade eder. 

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">SSE
</h1>


SSE = $\sum_{i=1}^{n} \varepsilon_i^2$

 Bu ifade her bir gözlem için hata terimlerini hesaplar ve bu hataların  karesini alıp toplar.
 
 SSE sonucumuz ne kadar düşük ise modelimizin daha performanslı olduğunu gösterir.
 
 Düşük hata modelin gözlemlerle uyumlu olduğunu gösterir.


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">Nasıl Yorumlanır ?
</h1>


 Eğer hata terimi pozitifse, bu, modelin gerçek değerden fazla tahmin ettiği anlamına gelir. Eğer negatifse, model gerçek değerden az tahmin etmiştir.
 
 Büyük hata terimleri, potansiyel olarak outlier'ları veya aykırı değerleri işaret edebilir. 

<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">MSE
</h1>


$\text{MSE (Mean Squared Error):} \quad MSE = \frac{1}{n} \sum_{i=1}^{n} (y_i - \hat{y}_i)^2$

${y_i}$ = Gerçek Değer

$\hat{y}_i$= Modelin Tahmin Ettiği Değer

n = Gözlem sayısı

###### Hataların Kareleri Ortalaması:

MSE, hataların karelerini alarak çalıştığı için büyük hataları vurgular. Kare alma işlemi, negatif ve pozitif hatalar arasındaki farkı düşürerek büyük hataları daha belirgin hale getirir.

###### Normalize Edilmemiş Değerler:

MSE, hataların karelerinin ortalaması olduğu için orijinal veri setinin biriminde ölçülüdür. Bu, sonuçların orijinal veri seti ile aynı birimde olmasını sağlar

###### Büyük Hatalara Duyarlılık:

MSE, büyük hatalara daha fazla duyarlıdır. Yüksek hata değerleri, MSE değerini daha fazla artırır.

MSE'nin karelerle çalıştığı ve aykırı değerlere duyarlı olduğu unutulmamalıdır!

<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">MAPE
</h1>


 MAPE (Mean Absolute Percentage Error), bir tahmin modelinin performansını değerlendirmek için kullanılan bir ölçüdür.
 
 MAPE, her gözlem için mutlak yüzde hatasının ortalaması alınarak hesaplanır. 


#### MAPE Nasıl Yorumlanır:

MAPE değeri yüzde cinsinden bir oranı temsil eder. Genellikle yüzde olarak ifade edilir. İşte MAPE değerinin yorumlanması:

0% MAPE: Mükemmel tahmin demektir. Modelin tahminleri, gerçek değerlerle tam olarak örtüşmektedir.

0% - 10% MAPE: Çok iyi bir tahmin modelini gösterir.

10% - 20% MAPE: Kabul edilebilir bir tahmin modelini gösterir.

20% - 50% MAPE: Orta seviyede bir tahmin modelini gösterir.

%50 ve üzeri MAPE: Düşük bir tahmin doğruluğunu gösterir ve genellikle kabul edilemez olarak değerlendirilir.

MAPE'nin dezavantajı, bazen sıfır olan gerçek değerlere bölme durumunda tanımsız olmasıdır.

<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">MAE
</h1>


$\text{MAE (Mean Absolute Error):} \quad MAE = \frac{1}{n} \sum_{i=1}^{n} \left| y_i - \hat{y}_i \right|$

$n$ : Gözlem sayısı

$y_i$ : Gerçek değerler

$\hat{y}_i$: Tahmmin edilen değerler

###### Aykırı Değerlere Duyarlılık:

MAE, hataların mutlak değerlerini kullanarak çalıştığı için aykırı değerlere daha dayanıklıdır. Yani, büyük hataların modelin performansını daha az etkilemesini sağlar.

##### MAE'nin düşük olması:
Düşük MAE değerleri modelin gerçek değere daha yakın tahminler yaptığını gösterir . Yani genellikle iyi bir tahmin yeteneğine sahip olduğunu gösterir.

<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">RMSE
</h1>


$\text{RMSE (Root Mean Squared Error):} \quad RMSE = \sqrt{MSE} $

###### RMSE'nin MSE'den farkları:
###### Aynı Ölçüm Birimi:

RMSE, MSE'nin karekökü alınmış hali olduğu için ölçüm birimi orijinal veri seti birimiyle aynıdır. Bu, RMSE'nin orijinal veri seti biriminde yorumlanabilir sonuçlar sağladığı anlamına gelir.

###### Büyük Hatalara Daha Fazla Hassasiyet:

RMSE, büyük hataları vurgulayan bir ölçüdür. Karekök alındığı için büyük hataların etkisi daha fazla olur. Bu durum, modelin genellikle büyük hatalardan kaçınmaya yönlendirilmesini sağlar.


###### MSE'ye Göre Daha Duyarlı:

RMSE, MSE'nin karekökü olduğu için aykırı değerlere daha fazla duyarlıdır. Yani, aykırı değerlerin modelin performansını daha fazla etkilemesine neden olabilir.

<h1 style="background-color:#FFA500;font-family:newtimeroman;font-size:200%;text-align:center;color:#FFFFFF">ANOVA
</h1>


ANOVA (Analysis of Variance), bir veya daha fazla grup arasında ortalamalar arasındaki istatistiksel anlamlılığı değerlendirmek için kullanılan bir istatistiksel yöntemdir. ANOVA, gruplar arasındaki varyansın, grup içi varyansa olan oranına dayanarak kararlar alır. Bu oran, gruplar arasındaki farklılıkların grup içi rastgele varyans ile karşılaştırılmasına dayanır.

ANOVA, özellikle gruplar arasında anlamlı bir fark olup olmadığını belirlemek ve çeşitli grupların ortalamaları arasındaki farklılıkları analiz etmek için kullanılır. ANOVA'nın temel amacı, gruplar arasındaki toplam varyansın, rastgele varyansa mı yoksa gerçek farklılıklara mı dayandığını belirlemektir.

ANOVA, genellikle bir bağımlı değişkenin kategorik bir bağımsız değişken tarafından nasıl etkilendiğini değerlendirmek amacıyla kullanılır. Örneğin, farklı tedavi yöntemlerinin bir hastalığın iyileşme süreci üzerindeki etkilerini değerlendirmek için ANOVA kullanılabilir.

ANOVA'nın temel iki türü vardır:

Tek Yönlü ANOVA (One-Way ANOVA): Bir bağımlı değişkenin kategorik bir bağımsız değişken tarafından etkilenip etkilenmediğini değerlendirir. Yani, tek bir bağımsız değişken içerir.

Çok Yönlü ANOVA (Two-Way ANOVA veya Factorial ANOVA): İki veya daha fazla bağımsız değişken içerir. Bu, bağımlı değişken üzerinde etkisi olan birden çok faktörü değerlendirmek için kullanılır.

ANOVA, gruplar arasındaki istatistiksel farklılıkları belirlemek ve hangi grupların birbirinden farklı olduğunu anlamak için kullanılan güçlü bir istatistiksel araçtır.

| Kaynaklar           | Toplam Kare | Serbestlik Derecesi | Ortalama Kare | F İstatistiği | p-değeri |
|---------------------|-------------|---------------------|---------------|---------------|----------|
| Model               | SSR         | k                   | MSR           | F             | p        |
| Hata (Residual)     | SSE         | n - k               | MSE           |               |          |
| Toplam (Total)       | SST         | n - 1               |               |               |          |


Toplam Kare (SSR): Modelin bağımlı değişkendeki toplam varyansı açıklama yeteneğini ölçer. Büyük bir SSR, modelin verileri daha iyi açıkladığı anlamına gelir.

Serbestlik Derecesi (k): Modeldeki bağımsız değişken sayısıdır. Daha fazla bağımsız değişken, modelin esnekliği artar.

Ortalama Kare (MSR): SSR'nin serbestlik derecesine bölünmüş halidir. Bu, modelin bağımlı değişken üzerindeki değişiklikleri açıklama yeteneğini ölçer.

F İstatistiği: MSR'nin MSE'ye bölünmesiyle elde edilir. Modelin istatistiksel olarak anlamlı olup olmadığını değerlendirmek için kullanılır.

p-değeri: F İstatistiği'nin istatistiksel anlamlılığını belirler. 0.05'den küçük bir p-değeri, modelin anlamlı bir şekilde farklılık gösterdiği anlamına gelir.


###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### Hata (Residual):

SSE: Model tarafından tahmin edilen değerlerle gerçek değerler arasındaki farkların karelerinin toplamını ölçer.

n - k: SSE'nin serbestlik derecesi, gözlemler arasındaki serbestlik derecesinden (n) modelin serbestlik derecesi çıkartılarak hesaplanır.

MSE: SSE'nin serbestlik derecesine bölünmüş halidir. Bu, modelin hatasının varyansını ölçer.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### Toplam (Total):

SST: Toplam varyansı ölçer. Bu, bağımlı değişkenin toplam varyansını ifade eder.

n - 1: SST'nin serbestlik derecesi, toplam gözlem sayısından bir çıkartılarak hesaplanır.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### ANOVA çıktısı
Eğer Model'in p-değeri anlamlıysa, en az bir grup diğerlerinden istatistiksel olarak farklıdır.

Eğer Model'in p-değeri anlamlı değilse, gruplar arasında anlamlı bir fark yoktur.

##### Modelin anlamlı olduğu durumda hangi grupların farklılık gösterdiğini belirlemek için post-hoc testler kullanılabilir

<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF"> Post-Hoc Testleri Nelerdir ?
</h1>


Post-hoc testler, bir ANOVA analizi sonrasında gruplar arasındaki özel farklılıkları belirlemek amacıyla kullanılan istatistiksel testlerdir. ANOVA analizi, gruplar arasında istatistiksel olarak anlamlı bir fark olduğunu gösterdiğinde, post-hoc testler hangi grupların birbirinden farklı olduğunu belirlemek için kullanılır. 

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### Tukey's Honestly Significant Difference (HSD)
Parametrik bir testtir. ANOVA sonrasında ortaya çıkan gruplar arasındaki farkları belirlemek için genellikle homojen varyans ve normal dağılım varsayımlarına dayanır.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### Duncan's Multiple Range Test: 
Parametrik bir testtir.Duncan'ın testi, önceki gruplar arasında anlamlı bir fark olup olmadığını belirlemek için kullanılır. ANOVA sonrasında genellikle homojen gruplar arasında kullanılır.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### Scheffe Test: 
Parametrik bir testtir ve genellikle büyük örneklem sayıları için uygundur. ANOVA sonrasında gruplar arasındaki farkları belirlemek için kullanılır.

<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF"> Parametrik olmayan (non-parametrik) Post Hoc Testleri
</h1>


#### Mann-Whitney U Testi (Wilcoxon Rank Sum Test): 

İki grup arasındaki farklılıkları değerlendirmek için kullanılır. Özellikle iki bağımsız grup arasında medyan farklılığı olup olmadığını test etmek için kullanılır.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### Wilcoxon İşaretli Sıralar Testi: 

İki eşleştirilmiş grubun arasındaki farklılıkları değerlendirmek için kullanılır. Bağımsız örneklemler arasındaki medyan farklılıklarını test etmek için de kullanılabilir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### Dunn'ın Testi:
Kruskal-Wallis testi sonrasında, hangi grupların farklı olduğunu belirlemek için kullanılır. İki grup arasındaki farkı değerlendiren Mann-Whitney U testini genişleterek, üç ya da daha fazla grup için uygulanabilir.


<h1 style="background-color:#FFA500;font-family:newtimeroman;font-size:200%;text-align:center;color:#FFFFFF">Doğrusal Regresyon Modelinin Özeti
</h1>


| **OLS Regression Results** |               | 
|---------------------------|---------------|
| **Dep. Variable:** Y       |               |
| **R-squared:** 0.750       |               |
| **Model:** OLS             |               |
| **Adj. R-squared:** 0.741  |               |
| **Method:** Least Squares   |               |
| **F-statistic:** 84.55     |               |
| **Date:** Tue, 13 Apr 2022  |               |
| **Prob (F-statistic):** 8.46e-11 |         |
| **Time:** 12:34:56          |               |
| **Log-Likelihood:** -74.028 |               |
| **No. Observations:** 50    |               |
| **AIC:** 154.1              |               |
| **Df Residuals:** 48        |               |
| **BIC:** 158.5              |               |
| **Df Model:** 1             |               |
| **Covariance Type:** nonrobust |            |

| **Coefficients** |     |               |
|------------------|-----|---------------|
| **const:** -5.6374 | std err: 3.543, t: -1.593, P>|t|: 0.118, [0.025, 0.975]: [-12.751, 1.476] | 
| **X1:** 1.2371    | std err: 0.134, t: 9.203, P>|t|: 0.000, [0.025, 0.975]: [0.967, 1.507]  |

| **Additional Statistics** |    |               |
|---------------------------|----|---------------|
| **Omnibus:** 0.073         |               |
| **Durbin-Watson:** 2.157   |               |
| **Prob(Omnibus):** 0.965   |               |
| **Jarque-Bera (JB):** 0.086|               |
| **Prob(JB):** 0.958        |               |
| **Skew:** 0.070            |               |
| **Kurtosis:** 2.819        |               |
| **Cond. No.:** 47.7        |               |


##### R-squared ve Adjusted R-squared:

R-squared: Bağımlı değişkenin varyasyonunun ne kadarının bağımsız değişkenler tarafından açıklandığını gösterir. Burada, R-squared değeri 0.75'tir, bu da modelin bağımlı değişkenin varyasyonunun %75'ini açıkladığını gösterir.

Adjusted R-squared: R-squared'ın düzeltilmiş bir versiyonudur, modeldeki ek bağımsız değişkenlerin eklenip eklenmediğini dikkate alır.
###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

##### F-statistic ve Prob (F-statistic):

F-statistic: Modelin genel anlamda anlamlılığını değerlendiren bir istatistiktir. Modeldeki bağımsız değişkenlerin modelin genel açıklama gücüyle karşılaştırılır.

Prob (F-statistic): F-statistic'in anlamlılık düzeyidir. Bu değer düşükse, model genel olarak anlamlıdır. Bu durumda, modelin genel anlamda anlamlılığını belirlemek için yeterli neden vardır.
###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

##### Katsayılar (Coefficients):

const (Kesme Noktası): Bu, bağımsız değişkenlerin değerleri sıfır olduğunda beklenen bağımlı değişkenin değeridir. Bu durumda, kesme noktası -5.6374'tür.

X1 (Bağımsız Değişken): Bu, X1'in her bir birim artışının bağımlı değişkendeki değişimi gösterir. Burada, X1'in katsayısı 1.2371'dir.
###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

#####  P>|t| ve [0.025, 0.975] İntervalleri:

P>|t|: Bu değer, bağımsız değişkenin katsayısının sıfır olma olasılığını ölçer. 0.05'ten küçükse, bağımsız değişkenin etkisi genellikle anlamlıdır.

[0.025, 0.975]: Bu aralık, bağımsız değişkenin katsayısının tahmin edilen değerinin güven aralığını gösterir. Eğer bu aralık sıfır içermiyorsa, bağımsız değişkenin etkisi genellikle anlamlıdır.
###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

#####  Additional Statistics:

Omnibus: Modelin normal dağılımı test eder. P değeri yüksekse, normal dağılım varsayımını karşılamaktadır.

Durbin-Watson: Modelin hata terimlerindeki otokorelasyonu ölçer. Değer 2'ye yakın olması istenir.

Prob(Omnibus): Omnibus istatistiğinin anlamlılık düzeyidir. 0.05'ten küçükse, normal dağılım varsayımını karşılamamaktadır.

Jarque-Bera (JB): Modelin normal dağılıma uygun olup olmadığını değerlendirir.

Prob(JB): JB istatistiğinin anlamlılık düzeyidir. 0.05'ten küçükse, normal dağılıma uygun değildir.

Skew ve Kurtosis: Modelin hata terimlerinin çarpıklık ve basıklık ölçümleridir.

Cond. No.: Varyans enflasyon faktörünü ölçer. Büyük değerler çoklu doğrusal bağlantı sorununa işaret edebilir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### AIC Nedir ?

AIC (Akaike Information Criterion):
Düşük Değerler İyi: AIC değeri ne kadar düşükse, modelin o kadar iyi uyum sağladığı kabul edilir.

Karşılaştırma: Farklı modeller arasında karşılaştırma yaparken, en düşük AIC değerine sahip model tercih edilir.

Karmaşıklık ve Uyum Dengesi: AIC, karmaşıklık ve uyum arasında bir denge kurmaya çalışır. Daha karmaşık modeller daha yüksek AIC değerlerine sahip olabilir, bu nedenle AIC, overfittinge karşı bir koruma sağlar.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### BIC Nedir ?

BIC (Bayesian Information Criterion):

Düşük Değerler İyi: BIC değeri de düşük olduğunda, modelin daha iyi uyum sağladığı kabul edilir.

Daha Katı Pénaltı: 

BIC, modelin karmaşıklığına daha katı bir pénaltı uygular. Bu nedenle, BIC daha az parametreye sahip basitleştirilmiş modelleri teşvik eder.

Popülasyon Seviyesinde İyi Performans: BIC, modelin popülasyon seviyesinde iyi performans gösterme olasılığını daha çok değerlendirir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


#### Nasıl Yorumlanır ?
AIC ve BIC değerleri modelin uygunluğunu değerlendirirken kullanılır.
Daha düşük değerler, daha iyi uyum sağladığına işaret eder.

AIC ve BIC'nin karşılaştırılması, farklı modeller arasında tercih yapmada yardımcı olabilir.

İki kriterin birbirine yakın olması durumunda, modelin genelleme yeteneği daha önemli bir kriterdir.

Örnek:

Model G'nin AIC değeri 500 ve Model H'nin AIC değeri 505 ise, Model G daha iyi bir uyum sağlar.

Burada, AIC değerleri arasındaki fark çok küçük olduğu için Model G'nin tercih edilmesi genellikle anlamlı bir iyileşme sağlamıyor olabilir.

Örnek:

Model I'nin AIC değeri 480 ve Model J'nin AIC değeri 470 ise, Model J daha iyi bir uyum sağlar.

AIC değerleri arasındaki fark daha belirgin olduğu için Model J'nin tercih edilmesi, bu durumda anlamlı bir iyileşme gösteriyor olabilir.

<h1 style="background-color:#3b5998;font-family:newtimeroman;font-size:300%;text-align:center;color:#FFFFFF">ÇOKLU LİNEER REGRESYON UYGULAMASI</h1>


In [1]:
import pandas as pd
from sklearn.linear_model import LinearRegression
import numpy as np
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm
import statsmodels.api as sm
import scipy.stats as stats


In [2]:
data = pd.read_csv("Marketing_Data.csv")

In [3]:
df = data.copy()

In [4]:
df

youtube  facebook  newspaper  sales
0      84.72     19.20      48.96  12.60
1     351.48     33.96      51.84  25.68
2     135.48     20.88      46.32  14.28
3     116.64      1.80      36.00  11.52
4     318.72     24.00       0.36  20.88
..       ...       ...        ...    ...
166    45.84      4.44      16.56   9.12
167   113.04      5.88       9.72  11.64
168   212.40     11.16       7.68  15.36
169   340.32     50.40      79.44  30.60
170   278.52     10.32      10.44  16.08

[171 rows x 4 columns]

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### YouTube: YouTube üzerinden yapılan harcamaları temsil eder.
###### Facebook: Facebook üzerinden yapılan harcamaları temsil eder.
###### Newspaper: Gazete reklamlarına yapılan harcamaları temsil eder.
###### Sales: Toplam satışları temsil eder.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


##### Bağımlı Değişken (y): Sales 

###### Bağımsız Değişkenler (X) : newspaper , youtube , facebook

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">Regresyon Modeli 
</h1>

In [5]:
X = df[['youtube', 'facebook', 'newspaper']]
y = df['sales']

model = LinearRegression()

model.fit(X, y)

coefficients = model.coef_
intercept = model.intercept_

print("Katsayılar (Coefficients):", coefficients)
print("Sabit (Intercept):", intercept)

Katsayılar (Coefficients): [0.04523544 0.18839783 0.00427949]
Sabit (Intercept): 3.5058709944650204


y = $\beta_0$ + $\beta_1$x$Youtube$ + $\beta_2$x$Facebook$ + $\beta_3$x$Newspaper$ 

y = $3.50$ + $0.04523544$ x $Youtube$ + $0.18839783$ x $Facebook$ + $0.00427949$ x $Newspaper$  

In [6]:
y_pred = model.predict(X)

In [7]:
df_hata = pd.DataFrame()

In [8]:
df_hata["Gercek_Degerler"] = y
df_hata["Tahmin"] = y_pred

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">Tahmin Değerleri
</h1>

In [9]:
df_hata

Gercek_Degerler     Tahmin
0              12.60  11.164980
1              25.68  26.025064
2              14.28  13.766342
3              11.52   9.275311
4              20.88  22.446400
..               ...        ...
166             9.12   6.486818
167            11.64   9.768662
168            15.36  15.249266
169            30.60  28.735611
170            16.08  18.093790

[171 rows x 2 columns]

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">Tahmin Hataları
</h1>

In [10]:
df_hata["Hata"] = df_hata["Gercek_Degerler"] - df_hata["Tahmin"]

In [11]:
df_hata

Gercek_Degerler     Tahmin      Hata
0              12.60  11.164980  1.435020
1              25.68  26.025064 -0.345064
2              14.28  13.766342  0.513658
3              11.52   9.275311  2.244689
4              20.88  22.446400 -1.566400
..               ...        ...       ...
166             9.12   6.486818  2.633182
167            11.64   9.768662  1.871338
168            15.36  15.249266  0.110734
169            30.60  28.735611  1.864389
170            16.08  18.093790 -2.013790

[171 rows x 3 columns]

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">SSE
</h1>

#### Her bir gözlem için modelin tahmin ettiği değer ile gerçek değer arasındaki farkların karelerini toplayarak hesaplanır

In [12]:
df_hata["SSE"] = df_hata["Hata"]**2

In [13]:
df_hata

Gercek_Degerler     Tahmin      Hata       SSE
0              12.60  11.164980  1.435020  2.059282
1              25.68  26.025064 -0.345064  0.119069
2              14.28  13.766342  0.513658  0.263845
3              11.52   9.275311  2.244689  5.038629
4              20.88  22.446400 -1.566400  2.453610
..               ...        ...       ...       ...
166             9.12   6.486818  2.633182  6.933645
167            11.64   9.768662  1.871338  3.501908
168            15.36  15.249266  0.110734  0.012262
169            30.60  28.735611  1.864389  3.475948
170            16.08  18.093790 -2.013790  4.055352

[171 rows x 4 columns]

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">MAPE
</h1>

In [14]:
df_hata['MAPE'] = (abs(df_hata['Hata'] / df_hata['Gercek_Degerler']) * 100)


In [15]:
df_hata

Gercek_Degerler     Tahmin      Hata       SSE       MAPE
0              12.60  11.164980  1.435020  2.059282  11.389048
1              25.68  26.025064 -0.345064  0.119069   1.343707
2              14.28  13.766342  0.513658  0.263845   3.597047
3              11.52   9.275311  2.244689  5.038629  19.485147
4              20.88  22.446400 -1.566400  2.453610   7.501917
..               ...        ...       ...       ...        ...
166             9.12   6.486818  2.633182  6.933645  28.872604
167            11.64   9.768662  1.871338  3.501908  16.076791
168            15.36  15.249266  0.110734  0.012262   0.720927
169            30.60  28.735611  1.864389  3.475948   6.092776
170            16.08  18.093790 -2.013790  4.055352  12.523572

[171 rows x 5 columns]

In [16]:
mape = df_hata['MAPE'].mean()
mape

14.076282723258855

0% MAPE: Mükemmel tahmin demektir. Modelin tahminleri, gerçek değerlerle tam olarak örtüşmektedir.

0% - 10% MAPE: Çok iyi bir tahmin modelini gösterir.

10% - 20% MAPE: Kabul edilebilir bir tahmin modelini gösterir.

20% - 50% MAPE: Orta seviyede bir tahmin modelini gösterir.

%50 ve üzeri MAPE: Düşük bir tahmin doğruluğunu gösterir ve genellikle kabul edilemez olarak değerlendiril

Mape değerimiz %14 çıktığı için kabul edilebilir bir tahmin olduğunu söyleyebiliriz

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">MAE
</h1>

In [17]:
df_hata['MAE'] = np.abs(df_hata['Gercek_Degerler'] - df_hata['Hata'])


In [18]:
df_hata.head().style

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">MSE
</h1>

In [19]:
df_hata['MSE'] = ((df_hata['Gercek_Degerler'] - df_hata['Tahmin']) ** 2).mean()


In [20]:
df_hata

Gercek_Degerler     Tahmin      Hata       SSE       MAPE        MAE  \
0              12.60  11.164980  1.435020  2.059282  11.389048  11.164980   
1              25.68  26.025064 -0.345064  0.119069   1.343707  26.025064   
2              14.28  13.766342  0.513658  0.263845   3.597047  13.766342   
3              11.52   9.275311  2.244689  5.038629  19.485147   9.275311   
4              20.88  22.446400 -1.566400  2.453610   7.501917  22.446400   
..               ...        ...       ...       ...        ...        ...   
166             9.12   6.486818  2.633182  6.933645  28.872604   6.486818   
167            11.64   9.768662  1.871338  3.501908  16.076791   9.768662   
168            15.36  15.249266  0.110734  0.012262   0.720927  15.249266   
169            30.60  28.735611  1.864389  3.475948   6.092776  28.735611   
170            16.08  18.093790 -2.013790  4.055352  12.523572  18.093790   

         MSE  
0    3.94527  
1    3.94527  
2    3.94527  
3    3.94527  
4    3.94527  
..       ...  
166  3.94527  
167  3.94527  
168  3.94527  
169  3.94527  
170  3.94527  

[171 rows x 7 columns]

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">RMSE
</h1>

In [21]:
df_hata['RMSE'] = np.sqrt(df_hata['MSE'])


In [22]:
df_hata.head().style

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FFA500;font-family:newtimeroman;font-size:200%;text-align:center;color:#FFFFFF">ANOVA
</h1>


In [23]:
model = ols("y ~ X", data).fit()
anova_results = anova_lm(model)
anova_results

df       sum_sq      mean_sq           F        PR(>F)
X           3.0  6103.984713  2034.661571  503.658112  2.089355e-83
Residual  167.0   674.641139     4.039767         NaN           NaN

<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">Youtube Değişkeni:

</h1>

$H_0$: youtube değişkenin bağımlı değişken üzerinde bir etkisi yoktur. (β1 = 0)

$H_1$ : youtube değişkeni, bağımlı değişken üzerinde anlamlı bir etkiye sahiptir. (β1 ≠ 0)

Tablodaki p-değeri (PR(>F)) oldukça küçüktür (3.262222e-73), bu da H0'ı reddetmemize olanak tanır. Bu nedenle, Youtube faktörü bağımlı değişken üzerinde anlamlı bir etkiye sahiptir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">Facebook Değişkeni:

</h1>

$H_0$: facebook değişkenin bağımlı değişken üzerinde bir etkisi yoktur. (β2 = 0)

$H_1$ : facebook değişkeni, bağımlı değişken üzerinde anlamlı bir etkiye sahiptir. (β2 ≠ 0)

Tablodaki p-değeri (PR(>F)) oldukça küçüktür (3.142090e-51), bu da H0'ı reddetmemize olanak tanır. Bu nedenle, facebook değişkeni bağımlı değişken üzerinde anlamlı bir etkiye sahiptir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">Newspaper Değişkeni:

</h1>

$H_0$: newspaper değişkenin bağımlı değişken üzerinde bir etkisi yoktur. (β3 = 0)

$H_1$ : newspaper değişkeni, bağımlı değişken üzerinde anlamlı bir etkiye sahiptir. (β3 ≠ 0)

Tablodaki p-değeri (PR(>F)) büyük olduğu için H0'ı reddedemeyiz. Bu nedenle, newspaper değişkeni bağımlı değişken üzerinde anlamlı bir bir etkiye sahip değildir.

<h1 style="background-color:#FFA500;font-family:newtimeroman;font-size:200%;text-align:center;color:#FFFFFF">Gruplar Arasındaki Farkı Test Etmek İçin İstatistiksel Testler
</h1>


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">Mann-Whitney U Testi (Wilcoxon Rank Sum Test):
</h1>

###### $H_0$ : İki grup arasında medyan farklılığı yoktur.
###### $H_1$ : İki grup arasında medyan farklılığı vardır.

In [24]:
statistic, p_value = stats.mannwhitneyu(df["youtube"], df["facebook"], alternative='two-sided')

print("Mann-Whitney U Test İstatistiği:", statistic)
print("P-değeri:", p_value)

alpha = 0.05
if p_value < alpha:
    print("İki grup arasında anlamlı bir medyan farklılığı vardır. (H0 reddedildi)")
else:
    print("İki grup arasında anlamlı bir medyan farklılığı yoktur. (H0 kabul edildi)")

Mann-Whitney U Test İstatistiği: 26749.0
P-değeri: 3.647283428087433e-40
İki grup arasında anlamlı bir medyan farklılığı vardır. (H0 reddedildi)


In [25]:
statistic, p_value = stats.mannwhitneyu(df["youtube"], df["newspaper"], alternative='two-sided')

print("Mann-Whitney U Test İstatistiği:", statistic)
print("P-değeri:", p_value)

alpha = 0.05
if p_value < alpha:
    print("İki grup arasında anlamlı bir medyan farklılığı vardır. (H0 reddedildi)")
else:
    print("İki grup arasında anlamlı bir medyan farklılığı yoktur. (H0 kabul edildi)")

Mann-Whitney U Test İstatistiği: 26176.5
P-değeri: 1.2748361032419896e-36
İki grup arasında anlamlı bir medyan farklılığı vardır. (H0 reddedildi)


In [26]:
statistic, p_value = stats.mannwhitneyu(df["newspaper"], df["facebook"], alternative='two-sided')

print("Mann-Whitney U Test İstatistiği:", statistic)
print("P-değeri:", p_value)

alpha = 0.05
if p_value < alpha:
    print("İki grup arasında anlamlı bir medyan farklılığı vardır. (H0 reddedildi)")
else:
    print("İki grup arasında anlamlı bir medyan farklılığı yoktur. (H0 kabul edildi)")

Mann-Whitney U Test İstatistiği: 16827.0
P-değeri: 0.01582186589532613
İki grup arasında anlamlı bir medyan farklılığı vardır. (H0 reddedildi)


###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FF00F7;font-family:newtimeroman;font-size:150%;text-align:left;color:#FFFFFF">Kruskal-Wallis Test İstatistiği ve Wilcoxon Testi
</h1>

In [27]:
import pandas as pd
from scipy.stats import kruskal, wilcoxon

kruskal_stat, kruskal_p = kruskal(df['facebook'], df['newspaper'], df['youtube'])

print("Kruskal-Wallis Test İstatistiği:", kruskal_stat)
print("Kruskal-Wallis Test P-değeri:", kruskal_p)

if kruskal_p < 0.05:  
    stat_facebook_newspaper, p_facebook_newspaper = wilcoxon(df['facebook'], df['newspaper'])
    print("\nFacebook vs. Newspaper:")
    print("Test İstatistiği:", stat_facebook_newspaper)
    print("P-değeri:", p_facebook_newspaper)
    
    stat_facebook_youtube, p_facebook_youtube = wilcoxon(df['facebook'], df['youtube'])
    print("\nFacebook vs. Youtube:")
    print("Test İstatistiği:", stat_facebook_youtube)
    print("P-değeri:", p_facebook_youtube)

    stat_newspaper_youtube, p_newspaper_youtube = wilcoxon(df['newspaper'], df['youtube'])
    print("\nNewspaper vs. Youtube:")
    print("Test İstatistiği:", stat_newspaper_youtube)
    print("P-değeri:", p_newspaper_youtube)
else:
    print("\nGruplar arasında anlamlı bir farklılık bulunmamaktadır.")


Kruskal-Wallis Test İstatistiği: 227.24553404036075
Kruskal-Wallis Test P-değeri: 4.510859128612107e-50

Facebook vs. Newspaper:
Test İstatistiği: 4999.0
P-değeri: 0.0002825355563991798

Facebook vs. Youtube:
Test İstatistiği: 264.0
P-değeri: 7.92213213450897e-28

Newspaper vs. Youtube:
Test İstatistiği: 325.0
P-değeri: 2.2253500690099148e-27


###### Facebook ve Newspaper grupları arasında anlamlı bir fark vardır. P-değeri, bu farkın tesadüfi olmadığını gösterir.

###### Facebook ve Youtube grupları arasında anlamlı bir fark vardır. P-değeri, bu farkın tesadüfi olmadığını gösterir.

######  Newspaper ve Youtube grupları arasında anlamlı bir fark vardır. P-değeri, bu farkın tesadüfi olmadığını gösterir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


<h1 style="background-color:#FFA500;font-family:newtimeroman;font-size:200%;text-align:center;color:#FFFFFF">Regresyon Analizi Özeti
</h1>


In [28]:
import statsmodels.api as sm
X_c = sm.add_constant(X) 
LR_OLS = sm.OLS(y,X_c)
LR_OLS_results = LR_OLS.fit()
LR_OLS_results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sales   R-squared:                       0.900
Model:                            OLS   Adj. R-squared:                  0.899
Method:                 Least Squares   F-statistic:                     503.7
Date:                Wed, 24 Jan 2024   Prob (F-statistic):           2.09e-83
Time:                        16:40:45   Log-Likelihood:                -359.99
No. Observations:                 171   AIC:                             728.0
Df Residuals:                     167   BIC:                             740.5
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          3.5059      0.400      8.772      0.000       2.717       4.295
youtube        0.0452      0.002     29.830      0.000       0.042       0.048
facebook       0.1884      0.009     20.895      0.000       0.171       0.206
newspaper      0.0043      0.007      0.658      0.511      -0.009       0.017
==============================================================================
Omnibus:                       57.934   Durbin-Watson:                   2.065
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              162.792
Skew:                          -1.388   Prob(JB):                     4.47e-36
Kurtosis:                       6.891   Cond. No.                         544.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Sabit Terim 

$H_0$: Sabit terim 0'a eşittir. (β0 = 0)

$H_1$ : Sabit terim 0'a eşit değildir. (β0 ≠ 0)

Tablodaki p-değeri (PR(|t|)) 0.05'den küçük olduğu için H0'ı red ederiz. Sabit terim model için anlamlıdır

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Youtube Değişkeni

$H_0$: youtube değişkeninin katsayısı 0'a eşittir. (β1 = 0)

$H_1$ : youtube değişkeninin katsayısı (β1 ≠ 0)

Tablodaki p-değeri (PR(|t|)) 0.05'den küçük olduğu için H0'ı red ederiz. Youtube değişkeni model için anlamlıdır

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Facebook Değişkeni

$H_0$: Facebook değişkeninin katsayısı 0'a eşittir. (β2 = 0)

$H_1$ : Facebook değişkeninin katsayısı (β2 ≠ 0)

Tablodaki p-değeri (PR(|t|)) 0.05'den küçük olduğu için H0'ı red ederiz. Youtube değişkeni model için anlamlıdır

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Newspaper Değişkeni

$H_0$: Newspaper değişkeninin katsayısı 0'a eşittir. (β3 = 0)

$H_1$ : Newspaper değişkeninin katsayısı (β3 ≠ 0)

Tablodaki p-değeri (PR(|t|)) 0.05'den büyük olduğu için H0'ı red edemeyiz. Newspaper değişkeni model için anlamlı değildir

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### R-kare (R-squared):

R-kare, bağımsız değişkenlerin bağımlı değişkenin değişkenliğini ne kadar açıkladığını gösteren bir ölçüdür. Bu durumda, R-kare değeri 0.900, satıştaki değişkenliğin yaklaşık %90'ının modeldeki bağımsız değişkenler tarafından açıklanabildiğini gösterir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Model:

OLS, Ordinary Least Squares'ın kısaltmasıdır ve bir lineer regresyon modelindeki parametreleri tahmin etmek için kullanılan bir yöntemdir. Bu yöntem, hataların karelerinin toplamını minimize eder.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Düzeltilmiş R-kare (Adjusted R-squared):

Düzeltilmiş R-kare, modeldeki tahmincilerin sayısını düzelttikten sonra R-kare'nin modifiye edilmiş bir versiyonudur. Bu, çoklu tahminciler olduğunda modelin uygunluğunu daha doğru bir şekilde ölçer. Bu durumda, 0.899 değerindedir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Yöntem (Method):

Least Squares, modelin katsayılarını tahmin etmek için kullanılan yöntemdir. Gözlemlenen ve tahmin edilen değerler arasındaki karelerin toplamını en aza indirir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### F-istatistiği (F-statistic):

F-istatistiği, modelin genel anlamda anlamlılığını test eder. Bu durumda, 503.7 değerindedir ve ilişkili olasılık (Prob (F-statistic)) sıfıra çok yakındır (2.09e-83), bu da genel modelin istatistiksel olarak anlamlı olduğunu gösterir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Log-likelihood:

Log-likelihood, modelin gözlemlenen veriyi ne kadar iyi açıkladığını ölçen bir değerdir. Bu durumda, -359.99 değerindedir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### AIC (Akaike Bilgi Kriteri) ve BIC (Bayesyen Bilgi Kriteri):

Bunlar istatistiksel modellerin göreli kalitesini ölçen değerlerdir. Düşük AIC ve BIC değerleri, daha iyi uyan bir modeli gösterir. Bu durumda, AIC 728.0 ve BIC 740.5 değerindedir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Serbestlik Dereceleri (Degrees of Freedom):

Df Residuals, kalıntılar için serbestlik dereceleri ve Df Model, model için serbestlik dereceleridir. Bu durumda, kalıntılar için 167, model için ise 3 serbestlik derecesi bulunmaktadır.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Kovaryans Türü (Covariance Type):

Bu, tahminde kullanılan kovaryans matrisinin türünü belirtir. Bu durumda "nonrobust" olarak belirtilmiştir, yani homoscedasticity ve kalıntılar arasında korelasyon olmadığını varsayar.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Omnibus Test:

H0 (Null Hypothesis): Modelin hata terimleri normal olarak dağılmıştır.

H1 (Alternative Hypothesis): Modelin hata terimleri normal olarak dağılmamaktadır.

Omnibus testi p-değeri 0.000 olarak verilmiş. Bu durumda, H0 hipotezi reddedilir. Yani, modelin hata terimleri normal olarak dağılmamaktadır.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Durbin-Watson İstatistiği:

Durbin-Watson istatistiği 2.065 olarak verilmiş. 2'ye yakın olması, otokorelasyon olup olmadığını belirlemede kullanılır. Bu durumda, otokorelasyon belirgin değil gibi görünüyor.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Jarque-Bera (JB) Testi:

H0: Hata terimleri normal olarak dağılmıştır.

H1: Hata terimleri normal olarak dağılmamaktadır.

Yorum: JB testi p-değeri 4.47e-36 (çok küçük) olarak verilmiş. Bu, H0 hipotezinin güçlü bir şekilde reddedildiğini gösterir. Yani, modelin hata terimleri normal olarak dağılmamaktadır.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Skewness( Çarpıklık )

0'dan küçükse: Dağılım sola çarpıktır .

0'a yakınsa: Dağılım simetriktir.

0'dan büyükse: Dağılım sağa çarpıktır .

Skew değeri -1.388	 olduğu için veri kümesi simetrik değildir. Sola çarpıktır

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Kurtosis (Basıklık)
Basıklık = 0: Dağılım normal dağılıma yakındır.
 
Basıklık < 0: Negatif bir değer, normal dağılıma göre daha düz bir dağılımı gösterir .
 
Basıklık > 0: Pozitif bir değer, normal dağılıma göre daha sivri bir dağılımı gösterir .
 
Basıklık değerimiz 6.891 olduğu için normal dağılıma göre daha sivri bir dağılımı gösterir.

###### _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ __ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


###### Condition Number

30'dan büyük bir değere sahip olduğunda multicollinearity yani çoklu bağlantı problemi ihtimali olduğunu söyleyebiliriz.